active/CornerRadius/CornerRadiusToThickness2.md

```diff
@@ -0,0 +1,149 @@
+
+# Background
+
+_This spec updates the existing [CornerRadiusToThicknessConverter](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.Primitives.CornerRadiusToThicknessConverter) class with new filters._
+
+WinUI has a [CornerRadius](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.CornerRadius) struct that's used to define the rounding the 4 corners of a box, and the [Thickness](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Thickness) struct that's used for defining the padding or margin around the 4 sides of a box.
+
+CornerRadius defines the radius length of the circle segment that's used to draw a rounded corner. For example the following uses 4 different radii for the 4 corners (the corners are listed in clockwise order starting from the top left corner):
+
+```xml
+<Border CornerRadius="0,20,40,60" BorderBrush="Red" BorderThickness="2" Height="100" Width="100" />
+```
+
+![Corner radius example](./CornerRadius.png)
+
+Thickness defines 4 sides. For example the following uses 4 different values for the margins around a blue rectangle in a red box (the sides are listed in clockwise order starting from the left):
+
+```xml
+<Border BorderBrush="Red" BorderThickness="2" Height="100" Width="100" >
+    <Rectangle Fill="Blue" Margin="0,20,40,60" />
+</Border>
+```
+
+![Thickness example](./Thickness.png)
+
+For consistency it's useful to be able to build a Thickness out of components of a CornerRadius. For this reason the [CornerRadiusFilterConverter](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.Primitives.CornerRadiusFilterConverter) exists, which can be used in Xaml markup. That converter has a [CornerRadiusFilterKind](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.Primitives.CornerRadiusFilterKind) that lets you define which parts to pull out, what's being added in this spec is update to that enum to allow more parts to be pulled out.
+
+**Existing values:**
+* FilterTopAndBottomFromLeft,
+* FilterTopAndBottomFromRight,
+* FilterLeftAndRightFromTop,
+* FilterLeftAndRightFromBottom
+
+**New values:**
+* FilterTopFromTopLeft,
+* FilterTopFromTopRight,
+* FilterRightFromTopRight,
+* FilterRightFromBottomRight,
+* FilterBottomFromBottomRight,
+* FilterBottomFromBottomLeft,
+* FilterLeftFromBottomLeft,
+* FilterLeftFromTopLeft,

```

Comment:
Looks like you can't mix them, so if you want to filter top and left, you're out of luck. (I'm okay with that, just checking.)

Reply:
Yes, you can't mix them. Maybe we should do this, but how should we do that? If we add more enum values for those cases, we would end up with quite a large enum.

Reply:
Discussion: It's not a [flags] enum, so it would be a breaking change to make it one now.